### PR TITLE
feat(authentication-service): add oauth-code read and write providers

### PR DESCRIPTION
**Pull request by akshatdubeysf**
## Description

Added providers to allow processing the code from Keycloak/Google authentications. Also removed username from subject of jwt tokens, and url of auth-redirects.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] Performed a self-review of my own code
- [X] Code conforms with the style guide

### CODE REVIEW

File: services/authentication-service/README.md [hunk 1]
```diff
@@ -152,6 +152,11 @@ Refer to [Database Migrations | LoopBack Documentation](https://loopback.io/doc/
 
 ### API Documentation
 
+
+### Providers
+
+You can find documentation for some of the providers bindings this service provides [here](./src/providers/README.md)
```
Comment (samarpan-b) [inline services/authentication-service/README.md:158]: You can find documentation for some of the providers available in this service [here]

File: services/authentication-service/src/modules/auth/login.controller.ts [hunk 2]
```diff
@@ -520,15 +533,15 @@ export class LoginController {
...
       };
-      const token = jwt.sign(codePayload, client.secret, {
-        expiresIn: client.authCodeExpiration,
-        audience: clientId,
-        subject: this.user.username,
-        issuer: process.env.JWT_ISSUER,
-      });
-      response.redirect(
-        `${client.redirectUrl}?code=${token}&user=${this.user.username}`,
+      const token = await keycloackCodeWriter(
```
Comment (samarpan-b) [inline services/authentication-service/src/modules/auth/login.controller.ts:536]: lets only keep one code writer provider. no need of separate for keycloak or google.

File: services/authentication-service/src/providers/README.md [hunk 3]
```diff
@@ -9,112 +9,169 @@ A [provider](http://loopback.io/doc/en/lb4/Creating-components.html#providers)
 is a class that provides a `value()` function. This function is called `Context`
 when another entity requests a value to be injected.
 
-Here we create a provider for a logging function that can be used as a new
-action in a custom [sequence](http://loopback.io/doc/en/lb4/Sequence.html).
+## Basic Usage
```
Comment (samarpan-b) [inline services/authentication-service/src/providers/README.md:12]: Mention that all these are optional.

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/authentication-service/README.md (hunk 1 above)
Comment (akshatdubeysf) [reply services/authentication-service/README.md]: done

File: services/authentication-service/src/providers/README.md (hunk 3 above)
Comment (akshatdubeysf) [reply services/authentication-service/src/providers/README.md:12]: Done

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**